test/e2e/autoscaling/custom_metrics_stackdriver_autoscaling.go

```diff
@@ -627,7 +630,7 @@ func ensureDesiredReplicasInRange(ctx context.Context, deploymentName, namespace
 		}
 	})
 	// The call above always returns an error, but if it is timeout, it's OK (condition satisfied all the time).
-	if wait.Interrupted(err) {
+	if wait.Interrupted(err) || errors.Is(err, fmt.Errorf("rate: Wait(n=%d) would exceed context deadline", 1)) {
```

Comment:
This `errors.Is` would never be true. Read [the official doc](https://pkg.go.dev/errors) to understand better.
https://go.dev/play/p/L1SuSGuUDCG

If you want to catch such errors with the specific message, `err.Error() == fmt.Sprintf("rate: Wait(n=%d) would exceed context deadline", 1)` is more correct. 
It'd be the best if we could catch error with errors.Is actually though, I'm not sure technical feasibility there, at least it'd take some effort I guess.

```go
package hoge_package

var DeadlineError := errors.New("rate: Wait(n=1) would exceed context deadline")

func WhereThisErrorOriginallyComesFrom(...) error {
  ...
  return DeadlineError
}

===

if wait.Interrupted(err) || errors.Is(err, hoge_package.DeadlineError) {} // It'll work as you expect.
```